qemu/tests/numa_prealloc_threads.py

```diff
@@ -51,19 +65,30 @@ def run(test, params, env):
     if expected_cpu_affinity != affinity:
         test.fail("Test and QEMU cpu-affinity does not match!")
 
-    cmd_taskset = "taskset -c -p " + str(thread_id)
-    output = process.getoutput(cmd_taskset)
-    if not re.search(affinity, output):
-        test.fail("The affinities %s and %s do not match!"
-                  % (affinity, str(output)))
+    cmd_taskset = "taskset -c -p %s" % thread_id
+    check_affinity(affinity, cmd_taskset, test)
 
     sandbox = params.get("qemu_sandbox", "on")
     if sandbox == "on":
         try:
             # The command is expected to fail
-            vm.monitor.qom_set(thread_context_device_id, "cpu-affinity", cpu_affinity)
+            vm.monitor.qom_set(thread_context_device_id,
+                               "cpu-affinity",
+                               cpu_affinity)
         except QMPCmdError as e:
             test.log.debug("The expected error message: %s and the output: %s"
                            % (error_msg, e.data))
             if not re.search(error_msg, str(e.data)):
-                test.fail("Can not get expected error message: %s" % error_msg)
+                test.fail("Cannot get expected error message: %s" % error_msg)
+
+    check_affinity(affinity, cmd_taskset, test)
+    cmd_taskset = "taskset -c -p %s %s" % (new_cpu_affinity, str(thread_id))
+    output = process.getoutput(cmd_taskset)
```

Comment:
It is better to print some execution steps so that it will be clearer in the debug log.
LOG.info("Set externally a new CPU affinity")

Reply:
@zhenyzha 
Updated now, thanks !